lib/llm/provider.rb

```diff
@@ -14,10 +14,10 @@ class Provider
     #  The host address of the LLM provider
     # @param [Integer] port
     #  The port number
-    def initialize(secret, host, port = 443)
+    def initialize(secret, **options)
```

Comment:
We could be more strict about what's accepted ? `host`, `port`, `ssl`.

Reply:
@0x1eef How?

Reply:
`LLM::Provider` could define `initialize` similar to this:
```ruby
def initialize(secret, host:, port: 443, ssl: true)
end
```
Then the subclasses of `LLM::Provider` could be similar to what you already have:
```ruby
def initialize(secret, **)
  # ...
end
```

And if you pass something like `blah`, that should be an error. What do you think?

Reply:
That’s what I thought initially, but i also then thought the options approach was fine

Reply:
It works - but then it's more like a Hash than keyword arguments. We can specify exactly which `keys` we accept, and if we're given some kind of rubbish - error out.